codersdk/workspaces.go

```diff
@@ -626,6 +626,34 @@ func (c *Client) UnfavoriteWorkspace(ctx context.Context, workspaceID uuid.UUID)
 	return nil
 }
 
+type ProvisionerTiming struct {
+	JobID     uuid.UUID `json:"job_id"`
+	StartedAt time.Time `json:"started_at"`
+	EndedAt   time.Time `json:"ended_at"`
+	Stage     string    `json:"stage"`
+	Source    string    `json:"source"`
+	Action    string    `json:"action"`
+	Resource  string    `json:"resource"`
+}
+
+type WorkspaceTimings struct {
+	ProvisionerTimings []ProvisionerTiming `json:"provisioner_timings"`
```

Comment:
Maybe add a TODO under here to add AgentScriptTimings, to explain why we have this parent struct to the reader.